helpers/kernel_config.go

```diff
@@ -174,53 +175,69 @@ var KernelConfigKeyIDToString = map[KernelConfigOption]string{
 type KernelConfig struct {
 	configs         map[KernelConfigOption]interface{} // predominantly KernelConfigOptionValue, sometimes string
 	needed          map[KernelConfigOption]interface{}
-	KConfigFilePath string
+	kConfigFilePath string
 }
 
 // InitKernelConfig inits external KernelConfig object
 func InitKernelConfig() (*KernelConfig, error) {
 	config := KernelConfig{}
 
 	// special case: user provided kconfig file (it MUST exist)
+
 	osKConfigFilePath, err := checkEnvPath("LIBBPFGO_KCONFIG_FILE") // override /proc/config.gz or /boot/config-$(uname -r) if needed (containers)
 	if err != nil {
-		return nil, err
+		return &config, err
 	}
 	if len(osKConfigFilePath) > 2 {
 		if _, err := os.Stat(osKConfigFilePath); err != nil {
-			return nil, err
+			return &config, err
 		}
-		config.KConfigFilePath = osKConfigFilePath // user might need to know used KConfig file in order to override kconfig for libbpf (example)
+		config.kConfigFilePath = osKConfigFilePath
 		if err := config.initKernelConfig(osKConfigFilePath); err != nil {
-			return nil, err
+			return &config, err
 		}
 
 		return &config, nil
 	}
 
 	// fastpath: check config.gz in procfs first
+
 	configGZ := "/proc/config.gz"
 	if _, err1 := os.Stat(configGZ); err1 == nil {
-		config.KConfigFilePath = configGZ // same thing here
+		config.kConfigFilePath = configGZ
 		if err2 := config.initKernelConfig(configGZ); err2 != nil {
-			return nil, err2
+			return &config, err2
 		}
 
 		return &config, nil
 	} // ignore if /proc/config.gz does not exist
 
 	// slowerpath: /boot/$(uname -r)
+
 	releaseVersion, err := UnameRelease()
 	if err != nil {
-		return nil, err
+		return &config, err
 	}
+
 	releaseFilePath := fmt.Sprintf("/boot/config-%s", releaseVersion)
-	config.KConfigFilePath = releaseFilePath // and here
-	if err := config.initKernelConfig(releaseFilePath); err != nil {
-		return nil, err
-	}
+	config.kConfigFilePath = releaseFilePath
+	err = config.initKernelConfig(releaseFilePath)
+
+	return &config, err
+}
+
+// GetKernelConfigFilePath gives the kconfig file chosen by InitKernelConfig during initialization
+func (k *KernelConfig) GetKernelConfigFilePath() string {
+	return k.kConfigFilePath
+}
+
+// AddCustomKernelConfigs allows user to extend list of possible existing kconfigs to be parsed from kConfigFilePath
+func (k *KernelConfig) AddCustomKernelConfigs(key KernelConfigOption, value string) error {
+	// extend initial list of kconfig options: add other possible existing ones
+	KernelConfigKeyIDToString[key] = value
+	KernelConfigKeyStringToID[value] = key
```

Comment:
What if the given key already exists (e.g. any number between 1-30 will collide with the already existing keys. Do we want to allow this? Warn? Error?

Also, it is not documented which number the user should choose for the key (I saw that in Tracee you used 1000 as the starting point arbitrarily).

Using strings as keys could have resolved this (as I suggested in a previous comment). If that is problematic to change, here is another suggestion:
1. Define a new constant for the starting point of custom keys (e.g. CUSTOM_CONFIG_START = 1000)
2. Enforce custom kernel configs to be bigger than this value and return error if not

WDYT?

Reply:
Addressed at: https://github.com/aquasecurity/libbpfgo/pull/73

About the strings, and not following that path from your previous comment, in some review I got another feedback to close the options (instead of having them opened as strings to be included). Nevertheless, I think it is working now and has better error handling than before the new PR mentioned. 

We can add an issue to improve this if it proves to be too bad by design. The way it is right now we can add the kconfig options and use them in tracee.bpf.c.